blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -430,6 +453,56 @@ private MetricData getStringMetricDataForRange(Locator locator, Range range, Gra
         return new MetricData(points, getUnitString(locator), MetricData.Type.STRING);
     }
 
+    public MetricData getEnumMetricDataForRange(final Locator locator, final Range range, Granularity gran) {
+        Map<Locator, MetricData> metricDataMap = getEnumMetricDataForRangeForLocatorList(new ArrayList<Locator>(){{add (locator); }}, range, gran);
+        return metricDataMap.get(locator);
+    }
+
+    public  Map<Locator, MetricData> getEnumMetricDataForRangeForLocatorList(final List<Locator> locator, final Range range, final Granularity gran) {
+        int enumThreadCount = Configuration.getInstance().getIntegerProperty(CoreConfig.ENUM_READ_THREADS);
+        final ColumnFamily<Locator, Long> CF = CassandraModel.getColumnFamily(RollupType.classOf(RollupType.ENUM, gran), gran);
+
+        ExecutorService taskExecutor = new ThreadPoolBuilder().withUnboundedQueue()
```

Comment:
I'm not sure you want to create a new ThreadPool each time this is called.  Isn't the purpose of a threadpool to have a set of threads warmed up and ready to go?